Create project-layout.php

maurodf0 opened the pull request:
Sorry first time contributing, hope i'm doing ok.

Add Project Layout

**Description**

Add the project layout pattern

maurodf0 committed:
Create project-layout.php

maurodf0 committed:
Update project-layout.php

maurodf0 committed:
Update project-layout.php

MaggieCabrera commented on patterns/project-layout.php:
```diff
@@ -0,0 +1,42 @@
+<!-- /wp:image -->
+
+<!-- wp:paragraph {"style":{"typography":{"fontSize":"0.8rem"}}} -->
+<p style="font-size:0.8rem"><?php echo esc_html__( "1. Queste basi mi hanno permesso di affrontare la sessione con un profondo apprezzamento per l'identità dell'hotel e una chiara visione della storia che volevo raccontare attraverso il mio obiettivo.", 'twentytwentyfour' ); ?></p>
```
I think the text here shouldn't have the `1.` at the start

maurodf0 replied:
The 1 is inside the new version on figma, shoud i delete it?

MaggieCabrera replied:
Sorry, I didn't see the new style on figma, I had to refresh my browser. It's fine to keep, but let's change the function to use `esc_html_x` and explain what this text is referring to. I would love if we could use the image caption for this, but I'm not sure that we can while maintaining the design

maurodf0 replied:
Yeah i thought about the caption but i think that without some custom css is impossible to have the same design of Figma.
Yeah, i'll use the `esc_html_x` thank u.

maurodf0 replied:
Hope now is ok

MaggieCabrera commented on patterns/project-layout.php:
```diff
@@ -0,0 +1,42 @@
+/**
+ * Title: Project Layout
+ * Slug: twentytwentyfour/project-layout
+ * Categories: text
```
let's add gallery to the list here

maurodf0 replied:
Ok, perfect. i didn't think about that ahah sorry.

maurodf0 replied:
Added.

MaggieCabrera commented on patterns/project-layout.php:
```diff
@@ -0,0 +1,42 @@
+
+?>
+
+<!-- wp:group {"style":{"spacing":{"padding":{"top":"6vw","right":"6vw","bottom":"6vw","left":"6vw"}},"elements":{"link":{"color":{"text":"#fffffffc"}}},"color":{"background":"#181818","text":"#fffffffc"}},"layout":{"type":"constrained"}} -->
```
This should have wide width. The colors should use the presets from the theme instead of the hex values and the spacing should also try to use the presets too

maurodf0 replied:
Thanks for all your help. I not founded the preset color, so i thought wasn't so important inside the all theme (i'll fix it).
How can i achive to use the preset for the spacing if not using the block spacing? 

Again, thanks for all your help

MaggieCabrera replied:
This is where you use the presets:

<img width="318" alt="Screenshot 2023-09-04 at 11 55 50" src="https://github.com/WordPress/twentytwentyfour/assets/3593343/e8cb9de2-dd68-415e-a88b-179accd3f283">

maurodf0 replied:
I think now is ok. Only a little question. There's no the same color of the background (based on figma file) i used the contrast color preset, is ok or i have to create the preset myself into the theme.json?)

Sorry for all the extra work i make you do...i'm trying to learn fastest possible for make no errors.

MaggieCabrera replied:
it's absolutely no issue! and thank you for following up on our reviews, that's what we are here for. You shouldn't need to create any presets, but chose what looks closer to the design. We have iterated away from some of the color to make them more accesible.

maurodf0 replied:
Thank u very much for the kind words.
Ok, i choosed the contrast color preset, the #222222 hope is ok for the pattern.

MaggieCabrera commented on patterns/project-layout.php:
```diff
@@ -0,0 +1,42 @@
+<!-- wp:column {"width":"33%"} -->
+<div class="wp-block-column" style="flex-basis:33%"><!-- wp:group {"layout":{"type":"flex","orientation":"vertical","verticalAlignment":"space-between","justifyContent":"stretch"}} -->
+<div class="wp-block-group"><!-- wp:paragraph {"style":{"typography":{"lineHeight":"1.2","fontSize":"1.4rem"}}} -->
+<p style="font-size:1.4rem;line-height:1.2"><strong><?php echo esc_html__( "La mia missione era tradurre la miscela unica di arte, artigianato e bellezza naturale dell'hotel in una raccolta di fotografie evocative che risuonassero con gli spettatori e trasmettessero lo spirito di questa incantevole destinazione. Mi sono avvicinato all'incarico con l'obiettivo di catturare il carattere dell'hotel  ...
```
This text shouldn't be bold

maurodf0 replied:
I used to the new screen into Figma, where u can see the new edit: https://www.figma.com/file/AlYr03vh4dVimwYwQkTdf6/Twenty-Twenty-Four?type=design&node-id=119-543&mode=design&t=ULaidtXPOPeoOe6Z-0

MaggieCabrera commented on patterns/project-layout.php:
```diff
@@ -0,0 +1,42 @@
+
+<!-- wp:group {"style":{"spacing":{"padding":{"top":"4vh"}}},"layout":{"type":"default"}} -->
+<div class="wp-block-group" style="padding-top:4vh"><!-- wp:paragraph {"style":{"typography":{"fontSize":"0.8rem"}}} -->
+<p style="font-size:0.8rem"><?php echo esc_html__( '2. Un racconto visivo di un ritiro intriso di arte, immagini che trascendono il regno delle semplici immagini.', 'twentytwentyfour' ); ?></p>
```
This text should be above the other one according to the design: https://github.com/WordPress/twentytwentyfour/issues/125

maurodf0 replied:
As the previous one, i referred to the new version in figma

maurodf0 committed:
Update project-layout.php

Used esc_html_x function
Added gallery category
Used preset for padding and colors (the bg color is not the same of the figma file because there's no preset like that color)

MaggieCabrera committed:
changed text and images, tweaks on markup

MaggieCabrera committed:
tweaked spacing, added aspect ratio and converted images

MaggieCabrera committed:
use preset font sizes

MaggieCabrera reviewed (approved):
I made a few changes to this one and rebased it. Changed the language to use the english version so it fits with the rest of the full page pattern, as well as the images so they make sense with the text changes too. I converted the images to webp and tweaked the markup to use presets where it wasn't using them

MaggieCabrera committed:
linting

maurodf0 commented:
> I made a few changes to this one and rebased it. Changed the language to use the english version so it fits with the rest of the full page pattern, as well as the images so they make sense with the text changes too. I converted the images to webp and tweaked the markup to use presets where it wasn't using them

Thanks for all your help. I'll see the edit u made so i'll make better the next time.
